src/tracks/InstrumentTrack.cpp

```diff
@@ -1148,6 +1174,29 @@ class fxLineLcdSpinBox : public LcdSpinBox
 			gui->fxMixerView()->setFocus();// set focus to fxMixer window
 			//engine::getFxMixerView()->raise();
 		}
+
+		virtual void contextMenuEvent( QContextMenuEvent* event )
+		{
+			// for the case, the user clicked right while pressing left mouse-
+			// button, the context-menu appears while mouse-cursor is still hidden
+			// and it isn't shown again until user does something which causes
+			// an QApplication::restoreOverrideCursor()-call...
+			mouseReleaseEvent( NULL );
+
+			QPointer<CaptionMenu> contextMenu = new CaptionMenu( model()->displayName(), this );
+
+			// This condition is here just as a safety check, fxLineLcdSpinBox is aways
+			// created inside a TabWidget inside an InstrumentTrackWindow
+			if ( InstrumentTrackWindow* window = dynamic_cast<InstrumentTrackWindow*>( (QWidget *)this->parent()->parent() ) )
```

Comment:
Well, one could make the argument that the menus should be pre-populated and stored as part of the InstrumentTrack object only to be updated at time to clicking, however, the creation and disposing of pop-up menus on-the-fly isn't all that uncommon (IIRC, QT does the memory disposal for us pretty well for these quickly created and disposed objects). 

In regards to the parent and casting, we have quite a bit of it in our code and so as long as we sanity check the type (which you've done) I have no room to criticize.

A more "kosher" solution may be to treat the pop-up as a singleton object and show/hide it as needed while updating its contents with the click event.

I may not be the best candidate for criticism here so I'd like to hear more from @Lukas-W and @diizy.